### PR TITLE
Changes to require slice date and time to be in ISO 8601 format #910

### PR DESCRIPTION
**Pull request by joachimmetz**
Changes to require slice date and time to be in ISO 8601 format #910

### CODE REVIEW

File: plaso/cli/helpers/event_filters.py [hunk 1]
```diff
@@ -35,12 +34,14 @@ def AddArguments(cls, argument_group):
...
         action='store', help=(
-            'Create a time slice around a certain date. This parameter, if '
-            'defined will display all events that happened X minutes before '
-            'and after the defined date. X is controlled by the parameter '
-            '--slice_size but defaults to 5 minutes.'))
+            'Date and time to create a time slice around. This parameter, if '
+            'defined, will display all events that happened X minutes before '
+            'and after the defined date, where X is controlled by the '
+            '--slice_size option, which is 5 minutes by default. The date and '
+            'time must be specified in ISO 8601 format including time zone '
```
Comment (Onager) [inline plaso/cli/helpers/event_filters.py:43]: Please add an example of what an ISO 8601 formatted date and time looks like.

File: plaso/cli/helpers/event_filters.py [hunk 2]
```diff
@@ -108,17 +109,22 @@ def ParseOptions(cls, options, configuration_object):
...
-      preferred_time_zone = getattr(
-          configuration_object, '_preferred_time_zone', None) or 'UTC'
-      timezone = pytz.timezone(preferred_time_zone)
-      time_slice_event_timestamp = timelib.Timestamp.FromTimeString(
-          time_slice_event_time_string, timezone=timezone)
-      if time_slice_event_timestamp is None:
+      if ' ' in time_slice_event_time_string:
         raise errors.BadConfigOption(
-            'Unsupported time slice event date and time: {0:s}'.format(
+            'Time slice date and time must be defined in ISO 8601 format.')
```
Comment (Onager) [inline plaso/cli/helpers/event_filters.py:114]: Please add an example of what an ISO 8601 formatted date and time looks like to this error message.

File: plaso/cli/helpers/event_filters.py [hunk 3]
```diff
@@ -108,17 +109,22 @@ def ParseOptions(cls, options, configuration_object):
...
+
+      date_time = dfdatetime_time_elements.TimeElements()
+
+      try:
+        date_time.CopyFromStringISO8601(time_slice_event_time_string)
+      except ValueError:
+        raise errors.BadConfigOption(
+            'Unsupported time slice date and time: {0:s}'.format(
```
Comment (Onager) [inline plaso/cli/helpers/event_filters.py:122]: I think it's worth adding an example of what the format is supposed to look like here as well, as this message is directly visible to the user.

File: plaso/cli/helpers/event_filters.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/cli/helpers/event_filters.py:43]: ack

File: plaso/cli/helpers/event_filters.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/cli/helpers/event_filters.py:122]: ack

**Comment by joachimmetz** [comment]
@Onager rebased and changes made, PTAL

**Pull request merged**